src/main/java/org/apache/pulsar/io/jcloud/sink/BlobStoreAbstractSink.java

```diff
@@ -179,32 +176,42 @@ private void flush() {
         } finally {
             rwlock.writeLock().unlock();
         }
-        final Iterator<Pair<Record<GenericRecord>, Blob>> iter = recordsToInsert.iterator();
-
-        while (iter.hasNext()) {
-            final Pair<Record<GenericRecord>, Blob> blobPair = iter.next();
-            try {
-                log.info("upload blob {}", blobPair.getLeft().getRecordSequence().get());
-                blobStore.putBlob(sinkConfig.getBucket(), blobPair.getRight(), PutOptions.NONE);
-                blobPair.getLeft().ack();
-                log.info("write success {}", blobPair.getLeft().getRecordSequence().get());
-            } catch (ContainerNotFoundException e) {
-                log.error("Bad message", e);
-                blobPair.getLeft().fail();
-                iter.remove();
-            }
+        Record<GenericRecord> firstRecord = recordsToInsert.get(0);
+        Schema<GenericRecord> schema = getPulsarSchema(firstRecord);
+        format.initSchema(schema);
```

Comment:
Yes, if fields are added, this may cause the new fields to be lost. If you do not initialize the schema at the beginning, let each record use its own schema, this will involve schema conversion and the problem of indeterminability of the parquet data format.

Of course, I can increase the schema version check. If the version changes, the current file will only be written here, and the data of the new schema will be output next time.